docs/ref/forms/models.txt

```diff
@@ -1,15 +1,246 @@
+      :attr:`~ModelFormOptions.fields` were set to ``'__all__'``) except for
+      those provided to :attr:`~ModelFormOptions.exclude`. 
+    * If neither of :attr:`~ModelFormOptions.fields` or
+      :attr:`~ModelFormOptions.exclude` are set, an
+      :class:`~django.core.exceptions.ImproperlyConfigured` exception will be
+      raised.
+
+``field_classes``
+-----------------
+
+.. attribute:: ModelFormOptions.field_classes
+
+    A dictionary that maps a model field name string to a
+    :class:`~django.forms.Field` class (**not** a string). 
+
+    :attr:`~ModelFormOptions.field_classes` can be used to set or
+    override the :class:`django.forms.Field` type for a field, including the
+    default form field associated with a given model :class:`django.db.models.Field` type.
+
+    The :class:`~django.forms.Field` specified in
+    :attr:`~ModelFormOptions.field_classes` (the 'new field') gets
+    passed all the arguments that would otherwise have been used for that
+    model's field (the 'old field'). If the old field has different attributes
+    (and therefore different arguments) than the new field, a :exc:`TypeError`
+    may be raised. For example:
+    
+    * Replacing a :class:`~django.forms.CharField` with an
+      :class:`~django.forms.EmailField` will work, since the latter is a
+      subclass of the former and can handle all the arguments passed to it.
+    * Replacing a :class:`~django.forms.CharField` with an
+      :class:`~django.forms.IntegerField`, however, will **not** work, since the
+      :class:`~django.forms.CharField` would pass a
+      :attr:`~django.forms.CharField.max_length` argument, and the
+      :class:`~django.forms.IntegerField` cannot accept such an argument (it
+      instead has :attr:`~django.forms.IntegerField.max_value`). In this case, a
+      :exc:`TypeError` will be raised with the message: 
+    
+        .. code-block:: console
+
+            TypeError: Field.__init__() got an unexpected keyword argument 'max_length'
+
+``fields``
+----------
+
+.. attribute:: ModelFormOptions.fields
+
+    One of :attr:`~ModelFormOptions.fields` or :attr:`~ModelFormOptions.exclude`
+    must be defined.
+
+    Valid values:
+
+    * The special string ``'__all__'`` — include every editable field on
+      :attr:`~ModelFormOptions.model` (non‑editable fields are always excluded).
+    * A tuple or list of distinct editable field name strings written in
+      snake_case (e.g. ``('title', 'author')`` or ``['title', 'author']``).
+
+    Any other input (a lone string that is not ``'__all__'``, a value that
+    is not a tuple/list, inclusion of invalid or non‑editable field names, or
+    omitting both :attr:`~ModelFormOptions.fields` and
+    :attr:`~ModelFormOptions.exclude`) will cause Django to raise an appropriate
+    exception (e.g. :exc:`TypeError`, :exc:`~django.core.exceptions.FieldError`,
+    or :exc:`~django.core.exceptions.ImproperlyConfigured`).
```

Comment:
```suggestion
    A tuple or list of :attr:`~ModelFormOptions.model` field names to be
    included in the form. The value ``'__all__'`` can be used to specify that
    all fields should be included.

    If any field is specified in :attr:`~ModelFormOptions.exclude`, this will
    not be included in the form despite being specified in
    :attr:`~ModelFormOptions.fields`.

    Either :attr:`~ModelFormOptions.fields` or
    :attr:`~ModelFormOptions.exclude` must be set. If neither are set, an
    :class:`~django.core.exceptions.ImproperlyConfigured` exception will be
    raised.
```